src/SHA1.elm

```diff
@@ -419,6 +485,12 @@ rotateLeftBy amount i =
         |> Bitwise.shiftRightZfBy 0
 
 
+rotateLeftBy1 : Int -> Int
+rotateLeftBy1 i =
+    -- because of how `rotateLeftBy1` is used, the `Bitwise.shiftRightZfBy 0` is not required
+    Bitwise.or (Bitwise.shiftRightZfBy 31 i) (Bitwise.shiftLeftBy 1 i)
```

Comment:
specializing to shift by 1 is important. This function takes only 1 argument now, an so elm does not need to apply wrappers that enable currying of this function. Much faster.